Alert - make Event to Check field required when adding/updating an Event Threshold Alert

himdel opened the pull request:
What the subject says, prevents a crash trying to show the alert when Event to Check is an empty array. 

https://bugzilla.redhat.com/show_bug.cgi?id=1204635

dclarizio commented:
@himdel Not sure what was supposed to be fixed, but when I try to add the alert described in the BZ (based on VM and Instance: Event Threshold, but leaving the actual parameters of the threshold unchosen), I now get the following exception and the record is Alert is not created:
```
[----] I, [2015-03-31T15:18:09.832905 #15882:971e90]  INFO -- : Started POST "/miq_policy/alert_edit?button=add" for 127.0.0.1 at 2015-03-31 15:18:09 -0700
[----] I, [2015-03-31T15:18:09.861874 #15882:971e90]  INFO -- : Processing by MiqPolicyController#alert_edit as JS
[----] I, [2015-03-31T15:18:09.861992 #15882:971e90]  INFO -- :   Parameters: {"button"=>"add"}
[----] F, [2015-03-31T15:18:10.018757 #15882:971e90] FATAL -- : Error caught: [NoMethodError] undefined method `empty?' for nil:NilClass
/home/dclarizio/dev/manageiq/vmdb/app/controllers/miq_policy_controller/alerts.rb:538:in `alert_valid_record?'
/home/dclarizio/dev/manageiq/vmdb/app/controllers/miq_policy_controller/alerts.rb:23:in `alert_edit'
```
Clicking on the old Alert I created recreating the bug itself, I still get the original exception `undefined method 'first' for nil:NilClass [miq_policy/tree_select] `.

Let me know if I'm not testing correctly.

himdel commented:
@dclarizio Oh, I hadn't thought of old alerts.. but it *is* a bug to have an Event Threshold Alert without an actual event, right? Adding a check to the template, but possibly we could want to add a migration that removes such alerts or something?

As for that exception .. that's interesting, seems sometimes `alert.expression != @edit[:new][:expression]` in there, I'll have to figure out which one we want to check.

dclarizio commented:
@himdel 
>but it is a bug to have an Event Threshold Alert without an actual event, right?

Correct, and I don't expect us to catch this error, I was just noting that the original symptoms had not gone away, so no need for a migration.

himdel commented:
Added a fix for displaying existing alerts without an event type.

Still missing a fix for that add/update should check for it.

himdel commented:
Hopefully done, added i18n to the template change and fixed the error you got while saving.

Also tested that it doesn't allow you to add or update an event threshold alert without the event type being set but doesn't crash when it isn't set.

dclarizio commented:
Verified fix.
